contracts/escrow/dapps/CompoundClaimCompFacet.sol

```diff
@@ -43,15 +46,19 @@ contract CompoundClaimCompFacet is PausableMods, DappMods {
         paused("", false)
         onlyBorrower(loanID)
     {
-        address escrow = address(LibEscrow.e(loanID));
+        bool loanUnavailable = LibLoans.loan(loanID).status >=
+            LoanStatus.Closed;
+        address user = loanUnavailable
+            ? msg.sender
+            : address(LibEscrow.e(loanID));
         LibEscrow.e(loanID).callDapp(
             address(COMPTROLLER_ADDRESS_PROVIDER_ADDRESS),
-            abi.encodeWithSignature("claimComp(address)", escrow)
+            abi.encodeWithSignature("claimComp(address)", user)
         );
 
-        // Add AAVE to escrow token list
-        // TODO: get the COMP token (add as immutable constructor variable)
-        LibEscrow.tokenUpdated(loanID, address(0));
+        if (loanUnavailable) {
+            LibEscrow.tokenUpdated(loanID, LibLoans.loan(loanID).lendingToken);
```

Comment:
we should rather be calling this function with the token addresses that we are updating (i.e. in this case `COMP`)

see my **TODO** comment that was removed in the diff above